### ISSUE TITLE
iOS: Download data freezes

### ISSUE THREAD

**Issue by criistii** [firebase/quickstart-unity#255]
I have a mobile game I am working on. I am currently storing all levels on Firebase for ease of development.
In this game I have a dedicated scene that handles contend download. My problem is that on iOS only, the 2nd time I download a level, the scene just freezes up. This does NOT happen the first time I load a level.

There is no error provided even after 10 minutes. This is my download code:

```
try
        {
            Debug.Log("!!!!!DownloadLevel " + levelId); //TODO: Crashes right after this line
            var data = await
                FirebaseDatabase
                .DefaultInstance
                .GetReference(DB_LEVEL)
                .Child(levelId)
                .GetValueAsync();
            Debug.Log("!!!!!DownloadLevel 1");

            var json = data.GetRawJsonValue();
            Debug.Log("!!!!!DownloadLevel 2");
            var level = JsonUtility.FromJson<Level>(json);
            Debug.Log("!!!!!DownloadLevel 3");
            return level;
        }
        catch (Exception ex)
        {
            Debug.LogError(ex);
            return null;
        }
```

My Xcode output looks like this:
!!!!!DownloadLevel c2192d6e-fb55-44d5-ba1f-55e1fe406fcf
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
FirebaseController:DownloadLevel(String)
FirebaseController:GetLevel(String)
LoadingScene:LoadLevel(String)
LoadingScene:LoadNextLevel(String)
LoadingScene:EvaluateActionProgress()
LoadingScene:Update()

As you can see, it never passes the GetValueAsync() function and it also does not reach  the catch{}

Please note that this is happening on iOS only. Both Android and the Unity Editor work just fine.

I am using Firebase 5.4.2. I event tried updating the pod dependencies to 5.13.0, but it didn't help.

Please advice. Thanks!

**Comment by criistii** [comment]
I added a verbose mode and got the following console output:

Firebase.Database.FirebaseDatabase (instance 0x11D587A00) registered for cleanup when 0x11D5CEAE0 is destroyed
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Firebase.Platform.FirebaseLogger:LogMessage(PlatformLogLevel, String)
Firebase.FirebaseApp:LogMessage(LogLevel, String)
Firebase.CleanupNotifier:RegisterForCleanup(IntPtr, IntPtr, Object, Type, DisposeObjectDelegate, DeleteObjectDelegate)
Firebase.Database.FirebaseDatabase:.ctor(FirebaseApp, ProxyReference)
Firebase.Database.FirebaseDatabase:GetInstance(FirebaseApp, String)
Firebase.Database.FirebaseDatabase:GetInstance(FirebaseApp)
Firebase.Database.FirebaseDatabase:get_DefaultInstance()
FirebaseController:DownloadLevel(String)
FirebaseController:GetLevel(String)
LoadingScene:LoadLevel(String)
LoadingScene:LoadNextLevel(String)
LoadingScene:EvaluateActionProgress()
LoadingScene:Update()

Not sure why the DB is destroyed. Any info?

**Comment by criistii** [comment]
I managed to fix this by updating from Unity 2018.1 to 2018.2

It seems that in Unity 2018.1 there is a bug that sometimes it just stops executing an async function. I managed to reproduce this at some point even in the editor.

After updating to 2018.2 the issue no longer appeared.
